provisioner/salt-masterless/provisioner.go

```diff
@@ -141,7 +142,9 @@ func (p *Provisioner) Prepare(raws ...interface{}) error {
 		return fmt.Errorf("Invalid guest_os_type: \"%s\"", p.config.GuestOSType)
 	}
 
-	p.guestCommands, err = guestexec.NewGuestCommands(p.config.GuestOSType, !p.config.DisableSudo)
+	// disable sudo here as we allow wrapping with execute_command to do
+	// dynamic sudo, the sudo provided by guestexec doesn't allow for -S
+	p.guestCommands, err = guestexec.NewGuestCommands(p.config.GuestOSType, false)
```

Comment:
This is a good improvement; another option would be to apply this patch to the SDK itself, (adding an optional ExecuteCommand to a new WrappedNewGuestCommands function where you allow the sdk consumer to pass in the ExecuteCommand